File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0 ||(x%10==0 && x!=0))
            return false;
        int reversenum=0,pop;
        while(x>reversenum){
            pop=x%10;
            reversenum=reversenum*10 + pop;
            x=x/10;
        }
        return x==reversenum || x== reversenum/10;
    }
}

File: 9-palindrome-number/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if((x<0) or ((x%10==0) and (x!=0))):
            return False
        
        reversedNum = 0
        while(x>reversedNum):
            pop = x%10
            reversedNum = reversedNum*10 + pop
            x=x//10
        return x==reversedNum or x==reversedNum//10
        
        